### ISSUE TITLE
Self hosting all the things

### ISSUE THREAD

**Issue by bertybuttface** [StanGirard/quiver#5]
I know this project is very reliant on cloud services in it's current state and for some people that may be fine.

Equally it may also now be possible to run this application using fully self hosted services.

* Whisper - https://github.com/guillaumekln/faster-whisper
* GPT - https://github.com/nomic-ai/gpt4all

Is this considered in or out of scope for this project? Would you accept a PR that provided local support?

**Comment by StanGirard** [comment]
Yes i would definetely accept a PR for local support. I want to build a desktop app in a few weeks/month - need time or help

**Comment by bertybuttface** [comment]
I agree a multi-platform app is a must. I'm a little out of touch with what the state of the art is but Electron sounds reasonable. I've not used Streamlit but it looks like a Firebase clone.

Decoupling the front and the backend is a possibility, would be nice to use the same frontend across web, mobile and desktop (via Electron) with an API built in Python. A single Docker container that runs the frontend and the API could be provided to keep setup simple.

As an aside there is a project called https://github.com/Xpitfire/symbolicai that has some overlap with this project which might be worth taking a look at. Won't solve the app issue but interesting to see what they are upto.

**Comment by bertybuttface** [comment]
Something like FastAPI (or Starlette) might be appropriate for the backend. Something like React / React Native for the frontend and maybe Electron for the mobile app. I could be showing my age here though.

**Comment by StanGirard** [comment]
Lets go with 

Python for the backend with FastAPI
NextJS for the frontend for web
Electron + NextJS for the Desktop app


I'll add that to the roadmap.

**Comment by bertybuttface** [comment]
This might be the easiest way to support GPT4All and others:

https://python.langchain.com/en/latest/modules/models/llms/integrations/gpt4all.html

**Comment by Klaudioz** [comment]
> Lets go with
> 
> Python for the backend with FastAPI NextJS for the frontend for web Electron + NextJS for the Desktop app
> 
> I'll add that to the roadmap.

I think basically pynecone is NodeJS with Python.

**Comment by StanGirard** [comment]
Closing in favor of #1. 

You can join the discussion on Discord or in the discussion panel 😉
